tasks.py

```diff
@@ -399,22 +401,45 @@ def check_migrations(context):
 
 @task(
     help={
-        "keepdb": "save and re-use test database between test runs for faster re-testing.",
-        "label": "specify a directory or module to test instead of running all Nautobot tests",
-        "failfast": "fail as soon as a single test fails don't run the entire test suite",
-        "buffer": "Discard output from passing tests",
-    }
+        "keepdb": "Save and re-use test database between test runs for faster re-testing.",
+        "label": "Specify a directory or module to test instead of running all Nautobot tests.",
+        "failfast": "Fail as soon as a single test fails don't run the entire test suite.",
+        "buffer": "Discard output from passing tests.",
+        "tag": "Run only tests with the specified tag. Can be used multiple times.",
+        "exclude_tag": "Do not run tests with the specified tag. Can be used multiple times.",
+        "verbose": "Enable verbose test output.",
+        "append": "Append coverage data to .coverage, otherwise it starts clean each time.",
+    },
+    iterable=["tag", "exclude_tag"],
 )
-def unittest(context, keepdb=False, label="nautobot", failfast=False, buffer=True):
+def unittest(
+    context,
+    keepdb=False,
+    label="nautobot",
+    failfast=False,
+    buffer=True,
+    exclude_tag=None,
+    tag=None,
+    verbose=False,
+    append=False,
+):
     """Run Nautobot unit tests."""
-    command = f"coverage run --module nautobot.core.cli test {label} --config=nautobot/core/tests/nautobot_config.py"
 
+    append_arg = " --append" if append else ""
+    command = f"coverage run{append_arg} --module nautobot.core.cli test {label} --config=nautobot/core/tests/nautobot_config.py"
     if keepdb:
         command += " --keepdb"
     if failfast:
         command += " --failfast"
     if buffer:
         command += " --buffer"
+    if verbose:
+        command += " --verbosity 2"
+    if tag is not None:
+        command += " --tag ".join([" "] + tag)
```

Comment:
Also not sure this is correct for multiple tags, I think maybe it should be:
```suggestion
        for individual_tag in tag:
            command += f" --tag {individual_tag}"
```
Same comment on the exclude_tag option as well.